### PR TITLE
Made-imagenette-path-configurable-in-config

### LINKED ISSUE #1832: Make imagenette path configurable in config
https://github.com/openvinotoolkit/anomalib/blob/34b3a90f9794dc037ec963381866d18f5f270328/src/anomalib/models/image/efficient_ad/lightning_model.py#L112

### PR DESCRIPTION
**Pull request by sahusiddharth**
## 📝 Description

- Provide a clear summary of the changes and the issue that has been addressed.
- 🛠️ Fixes #1832 

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](../docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: src/anomalib/models/image/efficient_ad/lightning_model.py [hunk 1]
```diff
@@ -62,6 +64,7 @@ class EfficientAd(AnomalyModule):
 
     def __init__(
         self,
+        imagenet_dir: Path | str = "./datasets/imagenette",
```
Comment (samet-akcay) [inline src/anomalib/models/image/efficient_ad/lightning_model.py:67]: @alexriedel1, a bit minor, but would you prefer this variable to be named `imagenet_dir` or `imagenette_dir`?

Comment (alexriedel1) [reply src/anomalib/models/image/efficient_ad/lightning_model.py:67]: `imagenet_dir` is a bit more comprehensive for future generations i guess

Comment (sahusiddharth) [reply src/anomalib/models/image/efficient_ad/lightning_model.py:67]: Hello @samet-akcay , @alexriedel1, how about `model_dir`? I would love to know your suggestions as well.

Comment (sahusiddharth) [reply src/anomalib/models/image/efficient_ad/lightning_model.py:67]: I think `penalty_loss_dataset_dir` is better, as it gives more information, @samet-akcay what do you think?

Comment (samet-akcay) [reply src/anomalib/models/image/efficient_ad/lightning_model.py:67]: If we choose `penalty_loss_dataset_dir`, we might need to rename some other variables as well such as 
https://github.com/openvinotoolkit/anomalib/blob/2c87edf81471b8591e725443bfac83b83878416e/src/anomalib/models/image/efficient_ad/torch_model.py#L356

If @alexriedel1 prefers `imagenet_dir` than `imagenette_dir`, then let's stick to `imagenet_dir` for now. If another refactor needed, we could work on it later on.

Thanks

**Review by samet-akcay — APPROVED** [review]
Thanks for addressing this

**Pull request merged**